### PR TITLE
MUS-219 PlayAnimation for multiple tacts

### PR DESCRIPTION
**Pull request by oliewa92**
Change the animation when pressing the play button and make it possible to play multiple tacts.

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/pocketmusic/ui/TactScrollRecyclerView.java [hunk 1]
```diff
@@ -52,11 +52,14 @@
 	private static final int MINIMUM_TACT_COUNT = 2;
 	private static final int TACT_VIEW_TYPE = 0;
 	private static final int PLUS_VIEW_TYPE = 1;
+	private static final int EMPTY_VIEW_TYPE = 2;
```
Comment (rohmab06) [inline catroid/src/main/java/org/catrobat/catroid/pocketmusic/ui/TactScrollRecyclerView.java]: I would assign TACTS_PER_SCREEN to it. Maybe a more specific name would be good: EMPTY_PLAY_ON_END_VIEW_TYPE?

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/pocketmusic/PocketMusicActivityTest.java [hunk 2]
```diff
@@ -128,10 +129,40 @@ protected boolean matchesSafely(TrackView item) {
 		};
 	}
 
+	private static Matcher<View> withRecyclerViewMeasuredSize() {
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/pocketmusic/PocketMusicActivityTest.java]: please rename to something describing what this does, eg. withSizeGreaterZero, (without looking at this matcher code i would have no idea what it is for when reading withRecyclerViewMeasuredSize in test code)

Comment (84n4n4) [reply catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/pocketmusic/PocketMusicActivityTest.java]: or better **is**SizeGreaterZero, to follow espresso matcher naming conventions.
"with" is used mostly for stuff that takes a parameter that describes something to look for, eg. withText, withId, etc.

while "is" is used for stuff that just matches a condition without parameters, isDisplayed()

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/pocketmusic/PocketMusicActivityTest.java [hunk 3]
```diff
@@ -128,10 +129,40 @@ protected boolean matchesSafely(TrackView item) {
...
+			}
+
+			@Override
+			protected boolean matchesSafely(TactScrollRecyclerView item) {
+				return item.getMeasuredHeight() > 0 && item.getMeasuredWidth() > 0;
+			}
+		};
+	}
+
+	private static Matcher<View> withRecyclerViewPlaying() {
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/pocketmusic/PocketMusicActivityTest.java]: same as above...

File: catroid/src/main/java/org/catrobat/catroid/pocketmusic/ui/TactScrollRecyclerView.java (hunk 1 above)
Comment (oliewa92) [reply catroid/src/main/java/org/catrobat/catroid/pocketmusic/ui/TactScrollRecyclerView.java]: Changed the name and the value of the types to keep the confusion to a minimum.

File: catroid/src/main/java/org/catrobat/catroid/pocketmusic/ScrollController.java [hunk 4]
```diff
@@ -33,58 +33,75 @@
...
 import org.catrobat.catroid.pocketmusic.note.NoteLength;
+import org.catrobat.catroid.pocketmusic.ui.AnimatorUpdateCallback;
+import org.catrobat.catroid.pocketmusic.ui.PianoView;
 import org.catrobat.catroid.pocketmusic.ui.TactScrollRecyclerView;
+import org.catrobat.catroid.pocketmusic.ui.TrackRowView;
 
 public class ScrollController {
 
-	private static final int LAST_NOTE_IN_TRACK_VIEW = 8;
+	private static final int NUMBER_OF_NOTES_IN_TRACK_VIEW = 8;
```
Comment (rohmab06) [inline catroid/src/main/java/org/catrobat/catroid/pocketmusic/ScrollController.java]: shouldn't it be dependend on the TACTS_PER_SCREEN variable from TactScrollRecyclerView?

Comment (oliewa92) [reply catroid/src/main/java/org/catrobat/catroid/pocketmusic/ScrollController.java]: you are correct, made it now dependent on TACTS_PER_SCREEN and the MusicalBeat of the TrackGrid in the TactScrollRecyclerView

Comment (rohmab06) [reply catroid/src/main/java/org/catrobat/catroid/pocketmusic/ScrollController.java]: great!

**Pull request merged**